### PR TITLE
[FEATURE] List profilers from DataContext

### PR DESCRIPTION
**Pull request by cdkini**
Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html).

Changes proposed in this pull request:
- List method and related tests

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/en/latest/contributing/style_guide.html?highlight=style%20guide)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html?highlight=checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/en/latest/contributing/testing.html#contributing-testing-writing-unit-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: great_expectations/data_context/data_context.py [hunk 1]
```diff
@@ -3251,6 +3251,16 @@ def run_checkpoint(
             **kwargs,
         )
 
+    def list_profilers(self) -> List[str]:
+        if self.profiler_store is None:
+            raise ge_exceptions.StoreConfigurationError(
+                f"Attempted to list profilers from a Profiler Store, which is not a configured store."
+            )
+        return profiler_toolkit.list_profilers(
```
Comment (NathanFarmer) [inline great_expectations/data_context/data_context.py:3259]: What is the `profiler_toolkit` and is it some sort of global?

**Review by NathanFarmer — COMMENTED** [review]
LGTM from a python perspective, just one question about our implementation.

File: great_expectations/data_context/data_context.py (hunk 1 above)
Comment (cdkini) [reply great_expectations/data_context/data_context.py:3259]: It's a utility file of sorts to encapsulate the behavior that is called in the DataContext. 

```python
from great_expectations.rule_based_profiler import toolkit # toolkit.py
```
I mirrored the implementation of Checkpoints: https://github.com/great-expectations/great_expectations/blob/develop/great_expectations/data_context/data_context.py#L3099-L3103

I mainly used it to adhere to existing patterns but I also like separating the logic out to simplify the ever growing DataContext.

Comment (Shinnnyshinshin) [reply great_expectations/data_context/data_context.py:3259]: @cdkini  the `toolkit` has already been helpful in the CLI-upgrade task that I've been working on :) Thank you for it

File: tests/rule_based_profiler/test_toolkit.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    keys = ["a", "b", "c"]
+    store.list_keys.return_value = [ConfigurationIdentifier(char) for char in keys]
+    res = list_profilers(store, ge_cloud_mode=False)
+
+    assert res == keys
+    assert store.list_keys.called
+
+
+@mock.patch("great_expectations.data_context.store.ProfilerStore")
+def test_list_profilers_in_cloud_mode(mock_profiler_store: mock.MagicMock):
```
Comment (Shinnnyshinshin) [inline tests/rule_based_profiler/test_toolkit.py:21]: Just curious, what is the difference between the `cloud_mode` test and the non-`cloud_mode` test?

**Review by Shinnnyshinshin — APPROVED** [review]
Thank you @cdkini one small question regarding the tests, but otherwise LGTM

File: tests/rule_based_profiler/test_toolkit.py (hunk 2 above)
Comment (cdkini) [reply tests/rule_based_profiler/test_toolkit.py:21]: Just testing both paths to ensure 100% coverage of the method.

When in cloud mode, `list_keys` returns a list of strings. In non-cloud mode, this is a list of `ConfigurationIdentifier` (which requires the extra step to extract the string value).

This is an implementation detail at the store level and mirrors the Checkpoint toolkit.

**Pull request merged**